### PR TITLE
feat(doc integrations): Create DocIntegration table

### PR DESCRIPTION
**Pull request by ceorourke**
Create a `DocIntegration` table to store document based integration data.

This is the first PR of a project to move these away from being hardcoded and into a db table - they will be created and managed in _admin rather than by making [code changes](https://www.notion.so/sentry/Spec-Doc-Integration-Management-d1fdbe4d2ca64066884f93fda08dfbab#dec746c8296342f292287e46c7f59ccb). See the spec [here](https://www.notion.so/sentry/Spec-Doc-Integration-Management-d1fdbe4d2ca64066884f93fda08dfbab#dec746c8296342f292287e46c7f59ccb) for more context.

### CODE REVIEW

File: src/sentry/migrations/0256_create_docintegration_table.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    ),
+                ),
+                ("name", models.CharField(max_length=255)),
+                ("slug", models.CharField(max_length=64, unique=True)),
+                ("author", models.CharField(max_length=255)),
+                ("description", models.TextField()),
+                ("url", models.URLField()),
+                ("popularity", models.PositiveSmallIntegerField(default=1, null=True)),
+                ("is_draft", models.NullBooleanField(default=True)),
+                ("metadata", sentry.db.models.fields.jsonfield.JSONField(null=True)),
```
Comment (markstory) [inline src/sentry/migrations/0256_create_docintegration_table.py:51]: Do you need any datetimes for when doc integrations were added or updated?

File: src/sentry/migrations/0256_create_docintegration_table.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            ],
+            options={
+                "db_table": "sentry_docintegration",
```
Comment (markstory) [inline src/sentry/migrations/0256_create_docintegration_table.py:54]: What kind of queries do you see being made to this table? Are there any columns that could benefit from having an index on them?

Comment (mgaeta) [reply src/sentry/migrations/0256_create_docintegration_table.py:54]: I think the main query is going to be index/search on the Integrations Directory, but there are probably going to ever be 20 rows max.

File: src/sentry/migrations/0256_create_docintegration_table.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/migrations/0256_create_docintegration_table.py:51]: No, I don't think so. This is purely an internal way for us to list integrations that other companies have made with us in the integration directory.

**Review by AniketDas-Tekky — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Review by wedamija — APPROVED** [review]

**Pull request merged**